src/App/App.js

```diff
@@ -14,10 +14,36 @@ class App extends Component {
         console.log(this.state.movies)
     }
 
+    componentDidMount = () => {
+        fetch("https://rancid-tomatillos.herokuapp.com/api/v2//movies/")
+        .then(res => {
+           if(!res.ok) {
+               throw new Error('Failed to fetch movies')
+           }
+           return res.json()})
+        .then(moviesData => this.setState({
+           movies: moviesData.movies
+        }))
+        .catch(error => {
+           console.error("There was a problem with the fetch", error)
+        })
+    }
+
     getMovieById = (id) => {
         const filteredMovie = this.state.movies.find(movie => movie.id === id)
-        this.setState({movie: filteredMovie})
-        console.log("filteredMovies", filteredMovie)
+
+        fetch(`https://rancid-tomatillos.herokuapp.com/api/v2//movies/${filteredMovie.id}/`)
+        .then(res => {
+           if(!res.ok) {
+               throw new Error('Failed to fetch the movie')
+           }
+           return res.json()})
+        .then(filteredMovieData => this.setState({
+           movie: filteredMovieData
+        }))
+        .catch(error => {
+           console.error("There was a problem with the fetch", error)
```

Comment:
I saw your note about wondering whether having this fetch call within the getMovieById() method is a good idea.  I want to leave that to y'all to determine as you keep working through your application.   Think about the way React works and the way users will use your application.  When are all the time that fetch call might end up being made (aka when are all the times that getMovieById would be invoked)?  Will it ever be invoked at a time when it's unnecessary to make a new fetch call?   Do you notice anything slow or buggy in your application that might be related to this method or the fetch call within it?   Do fetch calls _**have**_ to be tucked within a lifecycle method?

If the answer to all those is _no,_ it might be totally fine and good to have the fetch within that method! 

I would only change it if you can find a good reason to do so.